src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/FeatureSwitchManager.cs

```diff
@@ -441,7 +442,25 @@ public MethodIL GetMethodILWithInlinedSubstitutions(MethodIL method)
                 }
             }
 
-            return new SubstitutedMethodIL(method, newBody, newEHRegions.ToArray());
+            // Existing debug information might not match new instruction boundaries (plus there's little point
+            // in generating debug information for NOPs) - generate new debug information by filtering
+            // out the sequence points associated with nopped out instructions.
+            MethodDebugInformation debugInfo = method.GetDebugInfo();
+            if (debugInfo != null)
+            {
+                ArrayBuilder<ILSequencePoint> sequencePoints = new ArrayBuilder<ILSequencePoint>();
+                foreach (var sequencePoint in debugInfo.GetSequencePoints())
+                {
+                    if ((flags[sequencePoint.Offset] & OpcodeFlags.Mark) != 0)
```

Comment:
It is not unusual to see bad PDBs. Should we silently ignore out-of-range sequencePoints here?

Reply:
Yep, fixed!